frontend/src/upload_file.ts

```diff
@@ -155,7 +155,30 @@ class UploadFile {
     files.forEach(file => {
       const { fieldName, formId, renderer, uploads, uploadUrl } = this;
       const filename = file.name;
-      const uploadIndex = uploads.length;
+      let uploadIndex = uploads.length;
+
+      // #323 remove existing file
+      for (let index = 0; index < this.uploads.length; ++index) {
+        let existing_upload = this.uploads[index];
+        if (existing_upload instanceof Upload) {
+          if (existing_upload.file.name == filename) {
```

Comment:
I would use `===`

Reply:
I get a typescript error: `Property 'name' does not exist on type 'Blob | File ...`
Can be fixed by casing the `file` as a File:
```
if ((existing_upload.file as File).name == filename) {
```

Reply:
Used metadata